packages/core/src/provider/base.ts

```diff
@@ -221,15 +217,21 @@ export class SmartAccountProvider<
       throw new Error("transaction is missing to address");
     }
 
-    const _overrides: UserOperationOverrides = {};
-    if (overrides?.maxFeePerGas || request.maxFeePerGas) {
-      _overrides.maxFeePerGas = overrides?.maxFeePerGas ?? request.maxFeePerGas;
-    }
-    if (overrides?.maxPriorityFeePerGas || request.maxPriorityFeePerGas) {
-      _overrides.maxPriorityFeePerGas =
-        overrides?.maxPriorityFeePerGas ?? request.maxPriorityFeePerGas;
-    }
-    _overrides.paymasterAndData = overrides?.paymasterAndData;
+    const _overrides: UserOperationOverrides = {
+      maxFeePerGas:
+        overrides?.maxFeePerGas != null
+          ? overrides?.maxFeePerGas
+          : request.maxFeePerGas
+          ? fromHex(request.maxFeePerGas, "bigint")
+          : undefined,
```

Comment:
could you make a ticket to follow up and generalize this value prioritization ternary statement against request, overrides, and maxFeePerGas into a util method? Priority: P2 :)

I can imagine we might need this elsewhere in the future (like below).

Reply:
https://app.asana.com/0/1205598840815267/1206061917337242/f